code/game/objects/structures/mirror.dm

```diff
@@ -12,10 +12,17 @@
 	layer = ABOVE_WINDOW_LAYER
 	var/magical = FALSE
 
-/obj/structure/mirror/Initialize(mapload)
+/obj/structure/mirror/Initialize(mapload, dir, building)
 	. = ..()
 	if(icon_state == "mirror_broke" && !broken)
 		obj_break(null, mapload)
+	if(mapload)
+		return
+	if(dir)
+		setDir(dir)
+	if(building)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -30 : 30)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
```

Comment:
Use the defines for direction rather than magic numbers

Reply:
Oh alright, this was copy and pasted from code for other wall frames.